### ISSUE TITLE
Feat: Add a public_description field to workspaces

### ISSUE THREAD

**Issue by grignards** [tracim/tracim#3551]
## Feature description and goals

`ON_REQUEST` (and `OPEN`) spaces (see #3536) will need a short description intended for users who want to learn about the space contents.
This description should be kept separate from the current `description` as the current description has an arbitrary size and is not made to be displayed in a list.

- [ ] add a `public_description` field that is limited in length (200 characters?) **TBD**

## Which maximum length should we allow ?

@grignards I propose 200 characters, purely a "guess".

**Comment by inkhey** [comment]
About the length, here are some length we do currently use for user, to compare.
```
    MIN_PASSWORD_LENGTH = 6
    MAX_PASSWORD_LENGTH = 512
    MAX_HASHED_PASSWORD_LENGTH = 128
    MIN_PUBLIC_NAME_LENGTH = 3
    MAX_PUBLIC_NAME_LENGTH = 255
    MIN_EMAIL_LENGTH = 3
    MAX_EMAIL_LENGTH = 255
    MIN_USERNAME_LENGTH = 3
    MAX_USERNAME_LENGTH = 255
    MAX_IMPORTED_FROM_LENGTH = 32
    MAX_TIMEZONE_LENGTH = 32
    MIN_LANG_LENGTH = 2
    MAX_LANG_LENGTH = 3
    MAX_AUTH_TOKEN_LENGTH = 255
    MAX_RESET_PASSWORD_TOKEN_HASH_LENGTH = 255
```

**Comment by gbonaspetti** [comment]
> About the length, here are some length we do currently use for user, to compare.
> 
> ```
>     MIN_PASSWORD_LENGTH = 6
>     MAX_PASSWORD_LENGTH = 512
>     MAX_HASHED_PASSWORD_LENGTH = 128
>     MIN_PUBLIC_NAME_LENGTH = 3
>     MAX_PUBLIC_NAME_LENGTH = 255
>     MIN_EMAIL_LENGTH = 3
>     MAX_EMAIL_LENGTH = 255
>     MIN_USERNAME_LENGTH = 3
>     MAX_USERNAME_LENGTH = 255
>     MAX_IMPORTED_FROM_LENGTH = 32
>     MAX_TIMEZONE_LENGTH = 32
>     MIN_LANG_LENGTH = 2
>     MAX_LANG_LENGTH = 3
>     MAX_AUTH_TOKEN_LENGTH = 255
>     MAX_RESET_PASSWORD_TOKEN_HASH_LENGTH = 255
> ```
> 
> I don't know which size we should choose but 255 is widely used in tracim.

Maybe we can use the biggest value that we have, i.e. public_description limited to 512 characters. @grignards @inkhey what do you think?

**Comment by lebouquetin** [comment]
There is no reason to limit the length of this field. Limiting it is the way to have to upgrade the data model.

If we limit it to 512 and a user would like to put 513 he/she will be blocked and will rework hist text for no reason except "a technical decision without real means"

**Comment by inkhey** [comment]
> > About the length, here are some length we do currently use for user, to compare.
> > ```
> >     MIN_PASSWORD_LENGTH = 6
> >     MAX_PASSWORD_LENGTH = 512
> >     MAX_HASHED_PASSWORD_LENGTH = 128
> >     MIN_PUBLIC_NAME_LENGTH = 3
> >     MAX_PUBLIC_NAME_LENGTH = 255
> >     MIN_EMAIL_LENGTH = 3
> >     MAX_EMAIL_LENGTH = 255
> >     MIN_USERNAME_LENGTH = 3
> >     MAX_USERNAME_LENGTH = 255
> >     MAX_IMPORTED_FROM_LENGTH = 32
> >     MAX_TIMEZONE_LENGTH = 32
> >     MIN_LANG_LENGTH = 2
> >     MAX_LANG_LENGTH = 3
> >     MAX_AUTH_TOKEN_LENGTH = 255
> >     MAX_RESET_PASSWORD_TOKEN_HASH_LENGTH = 255
> > ```
> > 
> > 
> > I don't know which size we should choose but 255 is widely used in tracim.
> 
> Maybe we can use the biggest value that we have, i.e. public_description limited to 512 characters. @grignards @inkhey what do you think?

I was imaginating this public descriptions more as a small descriptions to know just a bit more about the workspace.
I do check https://www.blindtextgenerator.com/lorem-ipsum to compare and 512 is really nice if we do need a relatively long description, but depending how the ui decide to show it, it may be not the way to go.

Also, should this field be html or raw text ? If html this mean that many case should we clarify, if raw text, should we handle in some way the return to line ?

**Comment by raphj** [comment]
I agree with @lebouquetin. If we need to cut the description when displaying it, we can add ellipses ("…") or things like this (and maybe a notice in the UI if the description gets long).

**Comment by grignards** [comment]
> I agree with @lebouquetin. If we need to cut the description when displaying it, we can add ellipses ("…") or things like this (and maybe a notice in the UI if the description gets long).

Ok for that, but then what is the point of this description? Spaces already have a description (which can be very long/include HTML).

**Comment by lebouquetin** [comment]
> Ok for that, but then what is the point of this "short" description if it isn't short anymore? Spaces already have a description (which can be very long/include HTML) but we could display it with ellipsis (in fact this was my first idea: use the already existing space description).

The user interface must suggest to have a short description, this does not mean that technically it must be limited.

My point is:

- user interface should allow to write a one-line public description (which would be short). This is raw text
- user interface allow to write a long rich-text description (the current one + include tinyMCE in the textarea for ease of rich text writing)
- technically, both are stored as  unlimited text - this is not an issue.

**Comment by celiamgtt** [comment]
Did you already discuss where this public description will be displayed in the interface? I would need to know it to update the mockups :)

**Comment by inkhey** [comment]
> @grignards @lebouquetin Did you already discuss where this public description will be displayed in the interface? I would need to know it to update the mockups :)

As far i understood , the idea is that the public_description should be the only one returned in "Join space" list view.
I don't really know how both descriptions should be displayed in dashboard / workspace edition view. I do see public_workspace more as a "subtitile", so it should be displayed first (because the role is a short description).

**Comment by celiamgtt** [comment]
Thank you @inkhey, so it means that only the 'space manager' can see / edit the public description in the advanced dashboard for example?

**Comment by inkhey** [comment]
> Thank you @inkhey, so it means that only the 'space manager' can see / edit the public description in the advanced dashboard for example?

I do think rule to edit the public description should the same as for description. The only difference is that public description is visible to non-members through the "join space" list view.

As far i know the advanced dashboard is only accessible by admin or space manager, no change here.
